packages/date-time-picker/src/DateTimePicker.ts

```diff
@@ -0,0 +1,159 @@
+    }
+
+    public renderPicker(): TemplateResult {
+        const isDisabled = this.disabled || this.readonly;
+
+        return html`
+            <div class="picker">
+                ${this.renderStateIcons()}
+
+                <sp-picker-button
+                    class="picker-button"
+                    ?open=${this.open}
+                    ?quiet=${this.quiet || this.readonly}
+                    ?invalid=${this.invalid}
+                    ?disabled=${isDisabled}
+                    @click=${this.showPicker}
+                >
+                    <slot name="calendar-icon" slot="icon">
+                        <sp-icon-calendar></sp-icon-calendar>
+                    </slot>
+                </sp-picker-button>
+
+                <!-- TODO: Enable "receives-focus" when calendar is navigable via keyboard -->
+                <sp-overlay
+                    placement="bottom-start"
+                    receives-focus="false"
+                    .triggerElement=${this as HTMLElement}
+                    ?open=${this.open}
+                    @sp-closed=${this.hidePicker}
+                >
+                    <sp-popover class="popover">
+                        <div class="popover-content">
+                            <sp-calendar
+                                .selectedDate=${this.pickerDate}
+                                @change=${this.handleDate}
+                            ></sp-calendar>
+                        </div>
+                    </sp-popover>
+                </sp-overlay>
+            </div>
+        `;
+    }
+
+    public showPicker(): void {
+        this.pickerDate = this.getDateFromSegments();
+        this.open = true;
+    }
+
+    public hidePicker(): void {
+        this.setNewDateTime();
+        this.emitNewDateTime();
+
+        this.pickerDate = undefined;
+        this.open = false;
+    }
+
+    /**
+     * Updates the value of the internal property used to define the date used by the calendar with the value received
+     *
+     * @param event - Event with the value emitted by the calendar
+     */
+    private handleDate(event: CustomEvent<Date | undefined>): void {
+        event.stopPropagation();
+
+        const dateTime = event.detail;
+
+        if (dateTime) {
+            this.updateCurrentDate(dateTime);
+        }
+    }
+
+    /**
+     * Updates the segments using the `Date` object emitted by the calendar inside the picker
+     */
+    private updateCurrentDate(dateTime: Date): void {
+        this.pickerDate = dateTime;
+
+        const { year, month, day } = this.dateToCalendarDateTime(dateTime);
+
+        if (!this.yearSegment || !this.monthSegment || !this.daySegment) {
+            return;
+        }
+
+        this.yearSegment.value = year;
+        this.formatValue(this.yearSegment);
+
+        this.monthSegment.value = month;
+        this.formatValue(this.monthSegment);
+
+        this.daySegment.value = day;
+        this.formatValue(this.daySegment);
```

Comment:
can you not do these formatValue early and do it at the last execution statements? Much cleaner approach